include/s3select.h

```diff
@@ -622,7 +632,7 @@ struct s3select : public bsc::grammar<s3select>
     bsc::rule<ScannerT> muldiv_operator, addsubop_operator, function, arithmetic_expression, addsub_operand, list_of_function_arguments, arithmetic_argument, mulldiv_operand;
     bsc::rule<ScannerT> fs_type, object_path;
     bsc::rule<ScannerT> projections, projection_expression, alias_name, column_pos;
-    bsc::rule<ScannerT> when_case_else_projection, when_stmt;
+    bsc::rule<ScannerT> when_case_else_projection, when_case_else_projection_simple, when_stmt, when_stmt_simple;
     bsc::rule<ScannerT> const& start() const
```

Comment:
the syntax rules are not correct.
/when_case_else_projection_simple/ rule is "not connected" to root rule (start point)
/when_stmt_simple/ rule is not correct { arithmetic_expression >> bsc::str_p("when") } ,  means that expression before when could repeat more than once, which is wrong.